src/hooks/useAllowance.ts

```diff
@@ -116,11 +121,21 @@ export const useAllowance = ({
     }
   }, [approvalReceipt, isApprovalNeededQueryKey, resetApprove, queryClient]);
 
+  const isPending = (() => {
+    if (isApprovalNeeded) {
+      if (tokenAddress?.toLowerCase() === USDT_MAINNET_ADDRESS.toLowerCase()) {
+        return isPendingApproveConfigUsdtEth || isPendingAllowance;
+      }
+      return isPendingApproveConfigToken || isPendingAllowance;
+    } else return isPendingAllowance;
+  })();
```

Comment:
I'm still hesitant about IIFE in component body but I'm leaning to it, i think it's quite handy when we don't need `useMemo`. But lmk if you think it's not readable. We can change it to
```
let isPending;
if (...) {
isPending = ...
} else etc...
```